models/user.go

```diff
@@ -1026,6 +1026,19 @@ func GetUserByOpenID(id string) (*User, error) {
 	return user, nil
 }
 
+// GetUserByEmail returns the user object by given Email if exists.
+func GetUserByEmail(email string) (*User, error) {
+	user := &User{Email: email}
+	has, err := x.Get(user)
+	if err != nil {
+		return nil, err
+	} else if !has {
+		// TODO: look in UserEmail table (see user_mail.go)
+		return nil, ErrUserNotExist{0, email, 0}
+	}
+	return user, nil
+}
+
```

Comment:
@lunny how would this function look like if I wanted to also check in the UserMail table ?